src/clusterfuzz/_internal/bot/tasks/utasks/uworker_io.py

```diff
@@ -82,19 +82,43 @@ def get_entity_with_properties(ndb_key: ndb.Key, properties) -> ndb.Model:
   return entity
 
 
+def get_proto_fields(proto):
+  """Returns a generator containing tuples of (field_name, field_value) for each
+  field in the proto messsage. None is used for fields that are not set. This is
+  a little different than the behavior of protobuf which sets the value to a
+  Falsey version of the same type e.g. string fields are empty strings, int
+  fields are 0."""
+  for descriptor in proto.DESCRIPTOR.fields:
+    field_name = descriptor.name
+    try:
+      has_field = proto.HasField(field_name)
+    except AttributeError as error:
+      # This probably occurs after a deploy.
+      logs.log_error(f'Error getting proto fields {error}.')
+      has_field = False
+    if has_field:
+      field_value = getattr(proto, field_name)
+    else:
+      field_value = None
```

Comment:
Would it make sense to merge the `if-else` block into the `except` block?
E.g.,
```python
    try:
      proto.HasField(field_name)
      field_value = getattr(proto, field_name)
    except AttributeError as error:
      # This probably occurs after a deploy.
      logs.log_error(f'Error getting proto fields {error}.')
      field_value = None
```
or 
```python
    try:
      field_value = getattr(proto, field_name)
    except AttributeError as error:
      # This probably occurs after a deploy.
      logs.log_error(f'Error getting proto fields {error}.')
      field_value = None
```